src/sqlancer/mysql/gen/MySQLUpdateGenerator.java

```diff
@@ -0,0 +1,64 @@
+package sqlancer.mysql.gen;
+
+import java.sql.SQLException;
+import java.util.List;
+
+import sqlancer.Randomly;
+import sqlancer.common.gen.AbstractUpdateGenerator;
+import sqlancer.common.query.SQLQueryAdapter;
+import sqlancer.mysql.MySQLErrors;
+import sqlancer.mysql.MySQLGlobalState;
+import sqlancer.mysql.MySQLSchema.MySQLColumn;
+import sqlancer.mysql.MySQLSchema.MySQLTable;
+import sqlancer.mysql.MySQLVisitor;
+
+public class MySQLUpdateGenerator extends AbstractUpdateGenerator<MySQLColumn> {
+
+    private final MySQLGlobalState globalState;
+    private MySQLExpressionGenerator gen;
+
+    public MySQLUpdateGenerator(MySQLGlobalState globalState) {
+        this.globalState = globalState;
+    }
+
+    public static SQLQueryAdapter create(MySQLGlobalState globalState) throws SQLException {
+        return new MySQLUpdateGenerator(globalState).generate();
+    }
+
+    private SQLQueryAdapter generate() throws SQLException {
+        MySQLTable table = globalState.getSchema().getRandomTable(t -> !t.isView());
+        List<MySQLColumn> columns = table.getRandomNonEmptyColumnSubset();
+        gen = new MySQLExpressionGenerator(globalState).setColumns(table.getColumns());
+        sb.append("UPDATE ");
+        sb.append(table.getName());
+        sb.append(" SET ");
+        updateColumns(columns);
+        if (Randomly.getBoolean()) {
+            sb.append(" WHERE ");
+            MySQLErrors.addExpressionErrors(errors);
+            sb.append(MySQLVisitor.asString(gen.generateExpression()));
+        }
+        errors.add("doesn't have a default value");
```

Comment:
Some of these seem similar to the ones in https://github.com/sqlancer/sqlancer/blob/5bcd9ce69898f2866094535610ab0b4c04778843/src/sqlancer/mysql/gen/MySQLInsertGenerator.java#L91-L98. Could we extract the common expected errors and move them to https://github.com/sqlancer/sqlancer/blob/main/src/sqlancer/mysql/MySQLErrors.java?

Reply:
@eltfshr As we discussed in the meeting, can you check whether these errors are necessary? You can remove some expected errors that will not happen in the UPDATE statement, instead of copying them from INSERT statement.

Reply:
Yes, I checked each of errors, except "Data truncated for functional index". That one was a bug Manuel reported to MySQL, so I left it in. But on further analysis, I think it is not related to UPDATE. I will remove that one.

Reply:
Common expected errors are now extracted.